File: C#/Template/ImageCVReportGeneration.cs
using Template.Models;

namespace Template;

/// <summary>
/// Image Scanning for CV Report Generation
/// </summary>
public sealed class ImageCVReportGeneration : CvReportGeneration
{
    /// <summary>
    /// Image Scanning implementation
    /// </summary>
    /// <param name="cvPath"><inheritdoc/></param>
    /// <returns><inheritdoc/></returns>
    protected override Data ExtractData(string cvPath)
    {
        Console.WriteLine("Extracting data from image...");
        return new Data();
    }
}

File: typescript/template/image-cv-report-generation.ts
import { CVReportGeneration } from "./cv-report-generation";
import { Data } from "./data";
import { FileInfo } from "./file-info";

export class ImageCVReportGeneration extends CVReportGeneration {
  override extractData(file: FileInfo): Data {
    console.log("extracting data from image");
    return new Data();
  }
}
